src/pipeline.rs

```diff
@@ -69,20 +69,20 @@ pub fn build_graphs(
     let mut buffered = BufReader::new(input_file);
 
     let mut line = String::new();
-    let mut split: Vec<SmallVec<[String; SMALL_VECTOR_SIZE]>>;
     let mut parser = dom::Parser::default();
     while buffered.read_line(&mut line).unwrap() > 0 {
-        split = match &config.file_type {
-            FileType::JSON => read_json_columns(&line, &mut parser, &config.columns),
-
+        match &config.file_type {
+            FileType::JSON => {
+                let row = read_json_columns(&line, &mut parser, &config.columns);
+                entity_processor.process_row(&row);
+            }
             FileType::TSV => {
-              let row = line.trim().split('\t');
-              row.map(|c| c.split(' ').map(|s| String::from(s)).collect()).collect()
+                let row = line.trim().split('\t');
+                let row = row.map(|c| c.split(' ').collect()).collect();
```

Comment:
It seems odd to me to redefine a value with the same name on the next line. I would either give this a different name or put the whole thing on the same line.

Reply:
Actually, this is very common in rust world ;) 

See: https://doc.rust-lang.org/book/ch03-01-variables-and-mutability.html
"Shadowing thus spares us from having to come up with different names, such as spaces_str and spaces_num; instead, we can reuse the simpler spaces name."